typed-racket-lib/typed-racket/types/path-type.rkt

```diff
@@ -51,8 +56,10 @@
       ;; struct ops (non-prefab)
       [((Struct: _ _ flds _ _ _ _) (cons (StructPE: struct-ty idx) rst))
        #:when  (subtype t struct-ty)
-       (match-let ([(fld: ft _ _) (list-ref flds idx)])
-         (path-type rst ft (hash)))]
+       (handle-struct rst flds idx)]
+
+      [((Univ:) (cons (StructPE: struct-ty idx) rst))
```

Comment:
I think this should be something a little more general, so that it handles `(U Struct String)` instead of just `Any`.

Reply:
Ah Yes! Thanks!